site/gatsby-site/src/components/forms/SubmissionWizard/StepOne.js

```diff
@@ -115,7 +116,7 @@ const FormDetails = ({
     if (!values['date_downloaded']) {
       setFieldValue('date_downloaded', new Date().toISOString().substr(0, 10));
     }
-  }, []);
+  }, values['date_downloaded']);
```

Comment:
`useEffect` second param is an array, it looks like you want `[values.date_downloaded]` instead.